Portfolio: details popup window Added

mrcbq opened the pull request:
In this opportunity, the repo had:

🏁 When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
🏁 When the user clicks (or taps) the close (X) button, the popup disappears.
🏁 Using a JavaScript array to store all of the information for all projects. at a minimum: name, description, featured image, technologies, link to live version, link to the source.
🏁 Main page Updated. Now projects section is created dynamically using the information stored in that JavaScript object. all of the HTML in that section is created when the page loads.
🏁 Popup window for mobile and desktop versions implemented.

mrcbq committed:
Add data file

mrcbq committed:
Popup skeleton done

mrcbq committed:
Add icons to popup btns

mrcbq committed:
skeleton popup window end

mrcbq committed:
End logic for js file

mrcbq committed:
Modal window done

mrcbq committed:
Fix Linters errors

mrcbq committed:
Add popup function

Gardimy committed:
Update popup.js file

mrcbq committed:
Update the README file with the contributors and tech used

didierganthier commented on style.css:
```diff
@@ -531,3 +529,102 @@ footer {
+  color: #3c3a39;
+}
+
+.popup-container {
+  margin: 6% 7%;
+  padding: 24px;
+  border: 2px solid #ebecf0;
+  background-color: #fff;
+  position: fixed;
+  top: 100%;
+  bottom: 0;
+  left: 0;
+  right: 0;
+}
```
Well done on making the popup interaction 💯 but make sure to also adjust the design, I noticed some differences with the Figma 👇🏽 

## Figma (Desktop)

![Screenshot 2023-04-13 at 13 55 58](https://user-images.githubusercontent.com/45739963/231844040-5c1a7554-3c8d-4c84-aab0-e7faf3d12bc3.png)

## Your design (Desktop)

![Screenshot 2023-04-13 at 13 57 32](https://user-images.githubusercontent.com/45739963/231844432-9f963ec6-c00e-4057-971a-ebf8482f80f4.png)

My suggested changes for above ☝🏽  :
 - The popup image should be taking `55%` of the width or something approximating.
 - The technologies font size needs to also be bigger.
 - Lastly, I'd suggest adding some more `Lorem Ipsum` to manage the space better, blank space makes the design look incomplete a bit. 🤔 

Some suggested changes for the mobile design as well 👇🏽 

## Figma (mobile)

![Screenshot 2023-04-13 at 14 04 01](https://user-images.githubusercontent.com/45739963/231845761-0f805fb7-ec7a-4a85-b7ae-aa43769aea11.png)

## Your design (mobile, before I scroll)

![Screenshot 2023-04-13 at 13 56 57](https://user-images.githubusercontent.com/45739963/231844313-f55215b6-1195-46ce-acb1-e7cf776a429f.png)

## Your design (mobile, after I scroll)

![Screenshot 2023-04-13 at 13 56 43](https://user-images.githubusercontent.com/45739963/231844236-863582c1-1dd8-40c1-b15a-82617103a17b.png)

The first thing is that it's impossible for me to see the action buttons because there's some issue with the scrolling, and also the text goes outside of the popup, make sure that the scrolling (if there needs to be one) happens inside of the popup and that I can see the whole design. Keep going with the design 🥂 you're almost there 🤓

mrcbq replied:
Ok thanks for your recommendations, It's done.

didierganthier commented on README.md:
```diff
@@ -36,30 +36,32 @@ OPTIONAL SECTIONS:
 <!-- PROJECT DESCRIPTION -->
 
-# 📖 [setting a repo module 1] <a name="Portfolio Mobile first version"></a>
+# 📖 Portfolio Mobile First <a name="Portfolio Mobile first version"></a>
```
[OPTIONAL] Nice job using the README template, I'd suggest you avoid removing the `a` tags content, because they link each section to their right `Table of Contents` item, if you change the a tag content, make sure to also edit the `Table of Contents`

didierganthier reviewed (changes requested):
Hi Team,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.

<img src="https://media.giphy.com/media/3oEduMcvxTNPlGuhH2/giphy.gif" />

## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct GitHub flow ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

mrcbq commented on popup.js:
```diff
@@ -0,0 +1,359 @@
+  },
+];
+
+const worksContent = `<section class="works" id="works">
```
Ok thanks for your recommendations, It's done.

mrcbq commented on popup.js:
```diff
@@ -0,0 +1,359 @@
+</div>
+</section>`;
+
+const popupWindow = `    <div id="popup-container">
```
Ok thanks for your recommendations, It's done.

mrcbq commented on README.md:
```diff
@@ -36,30 +36,32 @@ OPTIONAL SECTIONS:
 <!-- PROJECT DESCRIPTION -->
 
-# 📖 [setting a repo module 1] <a name="Portfolio Mobile first version"></a>
+# 📖 Portfolio Mobile First <a name="Portfolio Mobile first version"></a>
```
Ok thanks for your recommendations, It's done.

NahnahAJ commented on popup.js:
```diff
@@ -0,0 +1,359 @@
+            </h3>
+            <p class="card-description inter400 font-white">
+                A daily selection of privately personalized reads; no accounts or
+                sign-ups required. Has been the industry's standard.
+            </p>
+            <ul class="tag">
+                <li class="tag-description-2">HTML</li>
+                <li class="tag-description-2">Bootstrap</li>
+                <li class="tag-description-2">Ruby</li>
+            </ul>
+        </div>
+        <button type="button" class="see-prj inter700 wide desktop-display-none">
+            See project
+        </button>
+    </div>
+
+
+    <div class="card-flex proArt card-type3 card6">
+        <div class="text-container card6-hidde">
+            <h3 class="card-title crete font-white desktop-display-none">
+                Profesional Art Printing Data
+            </h3>
+            <h3 class="card-title crete font-white display-none show-on-desktop">Website Portfolio</h3>
+            <p class="card-description inter400 font-white">
+                A daily selection of privately personalized reads; no accounts or
+                sign-ups required. Has been the industry's standard.
+            </p>
+            <ul class="tag">
+                <li class="tag-description-2">HTML</li>
+                <li class="tag-description-2">Bootstrap</li>
+                <li class="tag-description-2">Ruby</li>
+            </ul>
+        </div>
+        <button type="button" class="see-prj inter700 wide desktop-display-none">
+            See project
+        </button>
+    </div>
+
+</div>
+</section>`;
```
- Great job so far @Team :muscle: However, as the previous reviewer mentioned, the projects sections shouldn't be hardcoded. Good job removing the content from the html file but I noticed that you only replaced them in the javascript file using template literals.
- But as mentioned in the project requirements, the projects section needs to be created dynamically. :+1: I have attached a screenshot below.
![image](https://user-images.githubusercontent.com/102699958/231513024-d2bd2106-4953-4ce7-9436-9ec29140495c.png)

- [ ] This means the projects section should be created dynamically with the information from the projects array (all the HTML of the section is created when the page loads). This requires changes in existing code - already created HTML code for the projects list should be removed, and instead, it should be added via JavaScript. :wink: 
- You can check an example of this student's project: https://github.com/olawale-o/portfolio-mobile-version/pull/15/files#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R95
- So kindly use the information in the projects array and not hardcoded like you currently have.

NahnahAJ commented on popup.js:
```diff
@@ -0,0 +1,359 @@
+  const worksContainer = document.querySelector('#worksContainer');
+  worksContainer.innerHTML += worksContent;
+
+  const popupModal = document.querySelector('#popup-window');
+  popupModal.innerHTML += popupWindow;
+
```
- [ ] As the previous reviewer mentioned, in the desktop view, your popup styles still don't look completely like the original design. I noticed that the Image was smaller than that of the original design and the popup card looks smaller compared to that of the original design so kindly work on that.:wink:
- [ ] Aside from that, the buttons are not visible so kindly work on that as well :100: 
- Your Design
![image](https://user-images.githubusercontent.com/102699958/232118671-94d25263-6a06-45b2-9faf-faff6d7f6784.png)

- The Orignal Design
![image](https://user-images.githubusercontent.com/102699958/232118714-1c3174ae-adf1-47e8-b30b-bf92f7207ee9.png)


- [ ] In the mobile view, just as the previous reviewer mentioned, the rest of the paragraph text doesn't show as well as the buttons. So kindly fix this as well and ensue that your design looks similar to thye original design :+1: 

- The Original Design
![image](https://user-images.githubusercontent.com/102699958/232119623-f91f161b-7914-4a23-a27d-c18aa08dbad4.png)

- Your Design
[Screencast from 04-14-2023 05:45:43 PM.webm](https://user-images.githubusercontent.com/102699958/232119542-8a9213b5-ebc8-4fc4-96ac-65a62cdd5377.webm)

nahnahdev reviewed (changes requested):
## Project Status: CHANGES REQUIRED :recycle: 
Hi @mrcbq   and @Gardimy   , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.
<img src="https://media2.giphy.com/media/t8YPf3ACKeouLzEIvr/giphy.gif"/>

 ## Highlights
- Popup works as expected :heavy_check_mark:
- No linter errors :heavy_check_mark:
- Professional Readme :heavy_check_mark:

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mrcbq committed:
Correct all the CR suggestions:
Works section created dinamically trouight javascript
correct the popup window styles
other changes and bugs corrected

mrcbq committed:
Fix Linters errors

mrcbq commented:
> ## Project Status: CHANGES REQUIRED 
> Hi @mrcbq and @Gardimy ,
> 
> Great work on making the changes requested by a previous reviewer clap You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet. ![](https://camo.githubusercontent.com/a0bc459a4b2306bb32c4e43a2111f4a6e1c0fd304a1e8663241235b419387613/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966) [ ![68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966](https://camo.githubusercontent.com/a0bc459a4b2306bb32c4e43a2111f4a6e1c0fd304a1e8663241235b419387613/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966) ](https://camo.githubusercontent.com/a0bc459a4b2306bb32c4e43a2111f4a6e1c0fd304a1e8663241235b419387613/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966) [ ](https://camo.githubusercontent.com/a0bc459a4b2306bb32c4e43a2111f4a6e1c0fd304a1e8663241235b419387613/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966)
> 
> ## Highlights
> * Popup works as expected heavy_check_mark
> * No linter errors heavy_check_mark
> * Professional Readme heavy_check_mark
> 
> ## Required Changes 
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers, and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @NahnahAJ in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Sorry, I already updated the repository with the requested changes.  

🤦🏼 I didn't realize that I hadn't pushed the changes to the repository I was working on. I had basically left the repository intact because I was working on another repository, I pushed there and quickly left home.

Thanks for the second revision, I have worked very hard on the requested refactoring, I hope it is approved.

mrcbq commented:
Sorry, I already updated the repository with the requested changes.

🤦🏼 I didn't realize I hadn't pushed the changes to the repository I was working on yesterday. 
I had basically left the repository intact because I was working on another repository, I pushed there and quickly left home.

Thanks for the second revision, I have worked very hard on the requested refactoring, I hope it is approved.

ahmetbozaci commented:
### Hello @mrcbq 
### _This is an additional review as a result of the second opinion request._ 

- If you haven't pushed the changes, the second code reviewer is correct in asking you to do so. Only fill out the second opinion form, mentioned [here](https://github.com/mrcbq/portfolio-mobile-first/pull/9#:~:text=If%20you%20feel%20that%20the%20code%20review%20was%20unfair%2C%20you%20can%20request%20a%20second%20opinion%20using%20this%20form.), if you believe the code reviewer's feedback is unfair. The second opinion is not for an additional review. Please kindly request a review again from your dashboard.


___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

mrcbq commented:
Today has not been my day.
You can see that in the first review, I was asked for some changes, I did those yesterday all day, but when I submitted the second code review request I didn't update the changes properly in my repository but instead updated the changes in my code partner's repository.
I left the house in a hurry up way.

After it arrived, I continued working on the form validation in its validation-form branch and the local storage exercise in its respective localStorage branch.

At some point when I merged the changes from the other branches (validate-contact-form and popup ) what I had in the popup branch was merged and the pull request was closed.

however, the code of my main branch is updated with the requested changes.

I met with a few people who advised me that what I should do is try to explain the situation and submit the latest code review request.

I only have one Code review request left, please be patient with this review.

m05e5 reviewed (commented):
**Hi @mrcbq 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
![CleverScornfulBuffalo-size_restricted](https://user-images.githubusercontent.com/37901469/208869296-5ca9f6cc-a1b5-48fd-bd59-78fd83b35471.gif)

## Highlights


✔️ Clean Codebase
✔️ Nice UI
✔️ No linters error
✔️ Well done implementing the requested changes

## Optional suggestions

- [ optional ] Please next time just be careful not to merge a pull request until it's approved.
   You really did great. keep it up

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

Gardimy commented:
Hello Dear Marco,

How are you? Please don't worry about that. I know you made a great effort
to complete the projects. It is the same for me. But we have time, we can
always continue to fix the issues together.

Take care of yourself.

On Fri, Apr 14, 2023 at 11:13 PM Marco Baquero ***@***.***>
wrote:

> Today has not been my day.
> You can see that in the first review, I was asked for some changes, I did
> those yesterday all day, but when I submitted the second code review
> request I didn't update the changes properly in my repository but instead
> updated the changes in my code partner's repository.
> I left the house in a hurry up way.
>
> After it arrived, I continued working on the form validation in its
> validation-form branch and the local storage exercise in its respective
> localStorage branch.
>
> At some point when I merged the changes from the other branches
> (validate-contact-form and popup ) what I had in the popup branch was
> merged and the pull request was closed.
>
> however, the code of my main branch is updated with the requested changes.
>
> I met with a few people who advised me that what I should do is try to
> explain the situation and submit the latest code review request.
>
> I only have one Code review request left, please be patient with this
> review.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrcbq/portfolio-mobile-first/pull/9#issuecomment-1509483088>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2SPKRMOABLANEUCGKYQMUTXBIG45ANCNFSM6AAAAAAW5JLREU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
